### ISSUE TITLE
Console does not show the full log.

### ISSUE THREAD

**Issue by MaPePeR** [ATLauncher/ATLauncher#137]
I think when the log Messages come in very fast it drops a lot of them. (I could see a lot more log entries when comparing the log file to the internal log viewer)

This is really annoying when trying to troubleshoot something.

**Comment by RyanTheAllmighty** [comment]
Honestly no idea where to start with this. You may just be best to not use the console to debug since it's meh.

Any particular pack and version you notice this happen in so I can take a look and compare?

**Comment by MaPePeR** [comment]
There is no particular pack, that causes this,but it happened with code in ProjectE that I was testing. I wrote the result of some EMC mapping to the console. 

I tested this by adding code to the actionlistener on one of the image buttons, that starts a thread that spams a few lines into the log and I could reproduce it like that. 

(I spammed the console with a sawmill pattern made  from #-signs, calling the `LogManager.minecraft` method)

**Comment by MaPePeR** [comment]
The code i used to reproduce this (placed [here](https://github.com/ATLauncher/ATLauncher/blob/master/src/main/java/com/atlauncher/gui/components/BottomBar.java#L64-L69))

``` Java
        facebookIcon.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                LogManager.info("Opening Up ATLauncher Facebook Page");
                //Utils.openBrowser("http://www.facebook.com/ATLauncher");
                new Thread() {
                    @Override
                    public void run() {
                        String[] messages = {
                                "#",
                                "##",
                                "###",
                                "####",
                                "#####",
                                "######",
                                "#######",
                                "########",
                                "#########",
                                "##########"};
                        int i = 0;
                        while (i < 1000) {
                            LogManager.minecraft(String.format("%5d %s",i,  messages[i % messages.length]));
                            i++;
                        }
                    }
                }.start();
            }
        });
```

Example Output:

```
[23/2/2015 10:21:24 AM] Opening Up ATLauncher Facebook Page
[23/2/2015 10:21:24 AM]     0 #
[23/2/2015 10:21:24 AM]     1 ##
[23/2/2015 10:21:24 AM]     2 ###
[23/2/2015 10:21:24 AM]     3 ####
[23/2/2015 10:21:24 AM]     4 #####
[23/2/2015 10:21:24 AM]     5 ######
[23/2/2015 10:21:24 AM]     6 #######
[23/2/2015 10:21:24 AM]     7 ########
[23/2/2015 10:21:24 AM]     8 #########
[23/2/2015 10:21:24 AM]     9 ##########
[23/2/2015 10:21:24 AM]    10 #
[SNIP ---- Output is fine in between]
[23/2/2015 10:21:25 AM]   120 #
[23/2/2015 10:21:25 AM]   121 ##
[23/2/2015 10:21:25 AM]   122 ###
[23/2/2015 10:21:25 AM]   123 ####
[23/2/2015 10:21:25 AM]   124 #####
[23/2/2015 10:21:25 AM]   125 ######
[23/2/2015 10:21:25 AM]   126 #######
[23/2/2015 10:21:25 AM]   127 ########
[23/2/2015 10:21:25 AM]   128 #########
[23/2/2015 10:21:25 AM]   129 ##########
[23/2/2015 10:21:25 AM]   130 #
[23/2/2015 10:21:25 AM]   131 ##
[23/2/2015 10:21:25 AM]   134 #####
[23/2/2015 10:21:25 AM]   141 ##
[23/2/2015 10:21:25 AM]   205 ######
[23/2/2015 10:21:25 AM]   218 #########
[23/2/2015 10:21:25 AM]   252 ###
[23/2/2015 10:21:25 AM]   320 #
[23/2/2015 10:21:25 AM]   353 ####
[23/2/2015 10:21:25 AM]   385 ######
[23/2/2015 10:21:25 AM]   432 ###
[23/2/2015 10:21:25 AM]   454 #####
[23/2/2015 10:21:25 AM]   469 ##########
[23/2/2015 10:21:25 AM]   517 ########
[23/2/2015 10:21:25 AM]   542 ###
[23/2/2015 10:21:25 AM]   590 #
[23/2/2015 10:21:25 AM]   617 ########
[23/2/2015 10:21:25 AM]   630 #
[23/2/2015 10:21:25 AM]   643 ####
[23/2/2015 10:21:25 AM]   657 ########
[23/2/2015 10:21:25 AM]   671 ##
[23/2/2015 10:21:25 AM]   685 ######
[23/2/2015 10:21:25 AM]   696 #######
[23/2/2015 10:21:25 AM]   709 ##########
[23/2/2015 10:21:25 AM]   886 #######
[23/2/2015 10:21:25 AM]   898 #########
[23/2/2015 10:21:25 AM]   967 ########
```

it even causes problems for much smaller number of messages (300):

```
[SNIP ---]
[23/2/2015 10:25:30 AM]   120 #
[23/2/2015 10:25:30 AM]   121 ##
[23/2/2015 10:25:30 AM]   122 ###
[23/2/2015 10:25:30 AM]   123 ####
[23/2/2015 10:25:30 AM]   124 #####
[23/2/2015 10:25:30 AM]   125 ######
[23/2/2015 10:25:30 AM]   126 #######
[23/2/2015 10:25:30 AM]   127 ########
[23/2/2015 10:25:30 AM]   128 #########
[23/2/2015 10:25:30 AM]   129 ##########
[23/2/2015 10:25:30 AM]   130 #
[23/2/2015 10:25:30 AM]   131 ##
[23/2/2015 10:25:30 AM]   132 ###
[23/2/2015 10:25:30 AM]   135 ######
[23/2/2015 10:25:30 AM]   140 #
[23/2/2015 10:25:30 AM]   217 ########
[23/2/2015 10:25:30 AM]   224 #####
[23/2/2015 10:25:30 AM]   231 ##
[23/2/2015 10:25:30 AM]   239 ##########
[23/2/2015 10:25:30 AM]   244 #####
```

This is probably the case because you use the [`BlockingQueue.offer`](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html#offer(E)) Method, which only adds messages if it can so without blocking. I think using the [`BlockingQueue.put`](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html#put(E)) Method, which blocks the thread that wants to write into the queue should not be a problem. The generated Minecraft Log messages will probably pile up in the Stream, that is hooked to the stdout of Minecraft, but that *should* not be a problem.

Which brings me back to #138: 

To [quote s0cks](https://github.com/ATLauncher/ATLauncher/issues/138#issuecomment-74385064):
> yes there is an explanation for the main swing thread being blocked, if you understood concurrency in java you would know that when a blocking queue is full and inserted the thread that is doing the insertion gets blocked till it can insert in order to nullify potential memory leaks

So to sum up:

1. The thread that is doing the insertion is NOT the main gui thread, but [this anonymous Thread implementation](https://github.com/ATLauncher/ATLauncher/blob/d64fe39399bb32222b540026b71f2aca418172ac/src/main/java/com/atlauncher/data/Instance.java#L1091-L1138)
2. The thread that is doing the insertion does not use a method that blocks when the Queue is full.

So... What is causing the Main GUI thread to be locked then, again?

**Comment by RyanTheAllmighty** [comment]
This is not my area of knowledge so I can't say what will and won't help.

@s0cks wrote the code, she may have some insight into this.

Increasing the capacity of the queue helps solve the issue so maybe that would be fine? Otherwise using the put method also seems to work.

While I don't consider this mission critical, I do think this needs to be addressed as misordered messages in the case of fast logging is not something which is very ideal.

**Comment by MaPePeR** [comment]
Increasing the capacity is not really solving this issue. it only delays it.

The problem i see with this is, that Exceptions and Stacktraces are written as a big chunk of data, so the Queue will have a lot of incoming lines in a very short time. And you don't really want to risk loosing parts of a stacktrace.

**Comment by MaPePeR** [comment]
@RyanTheAllmighty: If @s0cks is not against that, i would ask you to change the logging code to use the blocking `put`-Method then.

**Comment by RyanTheAllmighty** [comment]
Alright this has caused way more issues than it's solved, it's being reverted

**Comment by MaPePeR** [comment]
For example?

**Comment by RyanTheAllmighty** [comment]
3 minute load times going up to 30 minute load times in high logging packs.

Whole thread of issues at http://forums.atlauncher.com/thread-8702.html

Regardless as soon as it was changed back, no issues.

Thanks for the input, but I'm going to leave the logging alone for the moment and not mess with it
